homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -134,15 +134,16 @@ def is_volume_muted(self):
     @property
     def supported_features(self):
         """Flag media player features that are supported."""
-        if self._mac:
-            return SUPPORT_VIERATV | SUPPORT_TURN_ON
         return SUPPORT_VIERATV
 
     def turn_on(self):
         """Turn on the media player."""
         if self._mac:
             self._wol.send_magic_packet(self._mac, ip_address=self._host)
-            self._state = STATE_ON
+        else:
+            # power button is actually power toggle
+            self._remote.turn_off()
```

Comment:
I really think you should PR the supporting library and add the turn_on() there... very strange the use of turn_off() inside turn_on()

Reply:
I agree, it looks strange to use "turn_off" inside "turn_on".

But as you can see in [https://github.com/florianholzapfel/panasonic-viera/blob/master/panasonic_viera/__init__.py](url):
`    def turn_off(self):
        """Turn off media player."""
        self.send_key(Keys.power)
`

"turn_off" just sends Keys.power, which in turn is defined as NRC_POWER-ONOFF and is just a toggle.

So you propose, I should do a PR on the viera library to duplicate the turn_off function?

Reply:
[https://github.com/florianholzapfel/panasonic-viera/pull/8](url)

Reply:
@dgomes OK, the PR got merged into master. Now we need [florianholzapfel](https://github.com/florianholzapfel) to create a new release and add that in requirements_all.txt and update my PR to "turn_on"?